### ISSUE TITLE
Unable to Build on Windows Server 2008 R2 with Visual studio 2015

### ISSUE THREAD

**Issue by ganeshran** [dotnet/coreclr#1328]
I am getting an error `No CMAKE_CXX_COMPILER found` on Windows Server 2008 R2 with Visual studio. Cmake is already added to the PATH variable and I am using the VS 2015 build switch too.

The cmake error logs say that WindowsSDKDir variable is missing.  I have not installed Windows SDK on the server. Is it a pre-requisite?

>
C:\Github\coreclr>build clean vs2015
Commencing CoreCLR Repo build
>
Doing a clean build
>
Could Not Find C:\Github\coreclr\bin\Logs\*_Windows_NT__x64__Debug.*
Checking pre-requisites...
>
Commencing build of native components for Windows_NT.x64.Debug
>
'"C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\Tools\\..\..\VC\vc
varsall.bat"' is not recognized as an internal or external command,
operable program or batch file.
-- The C compiler identification is unknown
-- The CXX compiler identification is unknown
CMake Error at CMakeLists.txt:5 (project):   No CMAKE_C_COMPILER could be found.
>
CMake Error at CMakeLists.txt:5 (project):  No CMAKE_CXX_COMPILER could be found.
>
-- Configuring incomplete, errors occurred!
See also "C:/Github/coreclr/bin/obj/Windows_NT.x64.Debug/CMakeFiles/CMakeOutput.
log".
See also "C:/Github/coreclr/bin/obj/Windows_NT.x64.Debug/CMakeFiles/CMakeError.l
og".
Failed to generate native component build project
>
C:\Github\coreclr>

**Comment by shahid-pk** [comment]
Are you running from a clean command prompt i mean don't use vs developer command prompt. In any case coreclr on vs2015 seems to be broken as this issue reports #1294

**Comment by ganeshran** [comment]
Yes I tried but the VSInstallDir variable wasn't set from a non Dev command prompt. It gives the DIA SDK error since the script is not able to determine where visual studio is installed.

This is a new Win 2008 R2 installation and I just installed .NET 4.5 and Visual Studio 2015 Trial version on it. After seeing the Cmake errors I also tried installing Windows SDK but that gives the same error.

Output from non developer tools command prompt

>C:\Github\coreclr>build clean vs2015
Commencing CoreCLR Repo build

>
Doing a clean build
>
Could Not Find C:\Github\coreclr\bin\Logs\*_Windows_NT__x64__Debug.*
Checking pre-requisites...
>
Commencing build of native components for Windows_NT.x64.Debug
>
'"C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\Tools\\..\..\VC\vc
varsall.bat"' is not recognized as an internal or external command,
operable program or batch file.
Error: DIA SDK is missing at "DIA SDK". This is due to a bug in the Visual Studi
o installer. It does not install DIA SDK at "" but rather at VS install location
 of previous version. Workaround is to copy DIA SDK folder from VS install locat
ion of previous version to "" and then resume build.
Visual Studio 2013 Express does not include the DIA SDK. You need Visual Studio
2013+ (Community is free).
See: https://github.com/dotnet/coreclr/blob/master/Documentation/project-docs/de
veloper-guide.md#prerequisites

**Comment by akoeplinger** [comment]
@ganeshran did you make sure you included the C++ components when you installed VS2015?

**Comment by ganeshran** [comment]
@akoeplinger I installed with the default options for VS Enterprise. Now modifying the install to include all components.

Will run the build again once install is done and report back

**Comment by ganeshran** [comment]
@akoeplinger 
It crossed the initial stage and started building but failed with 1150 warnings and 46 errors : 

Most of the errors are of the format : error C2220: warning treated as error - no 'object' file generated

**Comment by akoeplinger** [comment]
@ganeshran ok, that sounds like https://github.com/dotnet/coreclr/issues/1294.

**Comment by ganeshran** [comment]
Ah ok thanks @akoeplinger . Closing this issue since it's a duplicate of #1294
